src/fsharp/NameResolution.fs

```diff
@@ -1025,7 +1025,7 @@ let GetNestedTyconRefsOfType (infoReader: InfoReader) (amap: Import.ImportMap) (
 /// Handle the .NET/C# business where nested generic types implicitly accumulate the type parameters
 /// from their enclosing types.
 let MakeNestedType (ncenv: NameResolver) (tinst: TType list) m (tcrefNested: TyconRef) =
-    let tps = List.skip tinst.Length (tcrefNested.Typars m)
+    let tps = match tcrefNested.Typars m with [] -> [] | l -> List.skip tinst.Length l
```

Comment:
I'm curious why this only happens to enums? Wouldn't other types that are not generic who are enclosed by a generic type also have the same issue?

Reply:
Based off the comment I assumed nested types could use the enclosing types arguments and therefore implicitly inherited them whereas (another assumption on my part) I figured enums did not since there would be no use for them... With that said I tried https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGEA8AKgHyYDeqmV2WMALJgLIAUAlOZdQL6WdUCmAOwCuAW0wAhCPwAW5AAoB7RVwDcqVDxRA=

and it looks like the type param gets inherited for an enum as well. Inspecting Infer.Net further (see #11315) it looks like the actual issue might be that the enclosing type is defined in a non-generic class. So there's a `Variable` abstract class that defines the `Operator` enum and a `Variable<'T> :> Variable`. So if that's the case then the enum part is irrelevant and the issue is due to inheritance.